src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/ProfilerPass.php

```diff
@@ -41,6 +41,7 @@ public function process(ContainerBuilder $container)
             $priority = $attributes[0]['priority'] ?? 0;
             $template = null;
 
+            /** @var class-string|class-string<TemplateAwareDataCollectorInterface> $collectorClass */
```

Comment:
This annotation does not make sense. It's basically the same as…

```suggestion
            /** @var class-string $collectorClass */
```

In addition to this, I don't really understand which problem all those inline doc blocks solve. Please elaborate.

Reply:
If anything, we should add `@return class-string` to `Definition::getClass()` and revert all inline PHPdocs imho

Reply:
@derrabus Yeah, the reason was that my IDE had trouble understanding that `$collectorClass` could be a subclass of `TemplateAwareDataCollectorInterface`, but that's really an IDE issue I guess. In any case, inlining the `is_subclass_of` part seems to fix the issue, so hopefully that's ok.
@wouterj Done